qiskit_experiments/framework/base_experiment.py

```diff
@@ -348,7 +356,34 @@ def _run_jobs(self, circuits: List[QuantumCircuit], **run_options) -> List[Job]:
             job_circuits = [circuits]
 
         # Run jobs
-        jobs = [self.backend.run(circs, **run_options) for circs in job_circuits]
+        if self._use_sampler:
```

Comment:
Alternatively I would implement explicit `run_sampler` and `run_backend` method and dispatch based on the input `backend` type.